llama-index-core/llama_index/core/agent/runner/base.py

```diff
@@ -667,7 +667,10 @@ def stream_chat(
             chat_response = self._chat(
                 message, chat_history, tool_choice, mode=ChatResponseMode.STREAM
             )
-            assert isinstance(chat_response, StreamingAgentChatResponse)
+            assert isinstance(chat_response, StreamingAgentChatResponse) or (
```

Comment:
from my understanding, the chat response will either be StreamingAgentChatResponse or AgentChatResponse - the latter will occur if returna_direct is True right? 

imo is it hard to put everything in StreamingAgentChatResponse? seems a bit confusing to have two response types in `stream_chat`

Reply:
Yes that's right.

It was pretty hard to put everything into the one response type, since streaming responses are coupled with writing to memory and the llm response type (along with other factors)

It felt pretty obnoxious and messy to force an already streamed response into this abstraction 😅